concepts/generics/about.md

```diff
@@ -0,0 +1,93 @@
+# About
+
+## Introduction
+
+Generics are a language feature that enable code reuse.
+For example, it should not matter what type a list holds in order to compute its length.
+That is why the `length` function uses a generic type parameter `a` in its type signature.
+
+```elm
+length : List a -> Int
+```
+
+## Parameters in types
+
+When a type is generic over another type, we use a lowercase name for that other type, and call it a type parameter.
+Common examples are the `List`, `Maybe` and `Result` custom types.
+
+```elm
+-- Maybe is generic over the type a
+type Maybe a
+    = Nothing
+    | Just a
+
+someString : Maybe String
+someString = Just "hello"
+
+someInt : Maybe Int
+someInt = Just 42
+```
+
+A type can be generic over multiple other types, and thus have multiple type parameters.
+In that case, every independent type gets a different lowercase name.
+Being independent does not prevent them from becoming the same type, it just enables them to be different.
+
+```elm
+-- Result is generic over the types error and value
+type Result error value
+    = Ok value
+    | Err error
+```
+
+Record type alias' can also have type parameters, although it is less common.
+
+```elm
+-- Event is generic over the type a
+type alias Event a =
+    { at : Time
+    , data : a 
+    }
+```
+
+One important remark about names of type parameters, is that they can be multiple characters long.
+The only constraint is that they must be lowercase names.
+As such, when useful, we tend to give them descriptive names.
```

Comment:
I would move the 3 lines above line 31 "A type can be generic over multiple other types..." because we use a longer name for Result, it's weird to mention this after we used it already.